UI/ChatUI/ChatUI/ChatUI/MainPageViewModel.cs

```diff
@@ -0,0 +1,114 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.ComponentModel;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Windows.Input;
+using ChatUI.Entities;
+
+namespace ChatUI;
+
+public class MainPageViewModel : INotifyPropertyChanged
+{
+	private ObservableCollection<Message> _messages;
+	private string _currentMessage;
+	private bool _isMessageSent;
+
+	public event PropertyChangedEventHandler PropertyChanged;
+
+	public MainPageViewModel()
+	{
+		_messages = new ObservableCollection<Message>(Enumerable
+			.Range(0, 100)
+			.Select(i => new Message($"Message {i}", i % 2 == 0 ? "Me" : "Sender", DateTimeOffset.Now - TimeSpan.FromMinutes(30 * i), i % 2 == 0)));
+
+		AddMessage = new AddMessageCommand(ExecuteAddMessage);
+	}
+
+	public ObservableCollection<Message> Messages
+	{
+		get => _messages;
+		set
+		{
+			_messages = value;
+			OnPropertyChanged();
+		}
+	}
+
+	public string CurrentMessage
+	{
+		get => _currentMessage;
+		set
+		{
+			_currentMessage = value;
+			OnPropertyChanged();
+		}
+	}
+
+	public bool IsMessageSent
+	{
+		get => _isMessageSent;
+		set
+		{
+			_isMessageSent = value;
+			OnPropertyChanged();
+		}
+	}
+
+	public ICommand AddMessage { get; }
+
+	protected virtual void OnPropertyChanged([CallerMemberName] string propertyName = null)
+	{
+		PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(propertyName));
+	}
+
+	private void ExecuteAddMessage(string newMessageContent)
+	{
+		if(!string.IsNullOrEmpty(newMessageContent))
+		{
+			IsMessageSent = false;
```

Comment:
This sounds like a hack. Using (or "flashing") a `boolean` is not a valid structure for a "trigger", it should be an event. IMHO this should even remain in the view: either listen for `KeyPressed` event on the `TextBlox`, or listen for the `CollectionChanged` on the `ListView.Source` and scroll down on add.

Reply:
Removed, this is handled by `FeedView` now.